Pinta.Core/Classes/AsyncEffectRenderer.cs

```diff
@@ -44,24 +45,27 @@ internal abstract class AsyncEffectRenderer
 	internal sealed class Settings
 	{
 		internal int ThreadCount { get; }
-		internal int TileWidth { get; }
-		internal int TileHeight { get; }
+		internal RectangleI RenderBounds { get; }
+		internal bool EffectIsTileable { get; }
 		internal int UpdateMillis { get; }
 		internal ThreadPriority ThreadPriority { get; }
 
 		internal Settings (
 			int threadCount,
-			int tileWidth,
-			int tileHeight,
+			RectangleI renderBounds,
+			bool effectIsTileable,
 			int updateMilliseconds,
 			ThreadPriority threadPriority)
 		{
-			if (tileWidth < 0) throw new ArgumentOutOfRangeException (nameof (tileWidth), "Cannot be negative");
-			if (tileHeight < 0) throw new ArgumentOutOfRangeException (nameof (tileHeight), "Cannot be negative");
+			if (renderBounds.Width < 0) throw new ArgumentException ("Width cannot be negative", nameof (renderBounds));
+			if (renderBounds.Height < 0) throw new ArgumentException ("Cannot be negative", nameof (renderBounds));
 			if (updateMilliseconds <= 0) throw new ArgumentOutOfRangeException (nameof (updateMilliseconds), "Strictly positive value expected");
 			if (threadCount < 1) throw new ArgumentOutOfRangeException (nameof (threadCount), "Invalid number of threads");
-			TileWidth = tileWidth;
-			TileHeight = tileHeight;
+
+			// If the effect isn't tileable, there is a single tile for the entire render bounds.
```

Comment:
This comment doesn't really make sense here - maybe it should be lower down around the call to VerticalSliceup()?